experimental/gittuf/root.go

```diff
@@ -776,6 +782,166 @@ func (r *Repository) RemovePropagationDirective(ctx context.Context, signer ssli
 	return r.updateRootMetadata(ctx, state, signer, rootMetadata, commitMessage, options.CreateRSLEntry, signCommit)
 }
 
+// AddHook defines the workflow for adding a file to be executed as a hook. It
+// writes the hook file, populates all fields in the hooks metadata associated
+// with this file and commits it to the root of trust metadata.
+func (r *Repository) AddHook(ctx context.Context, signer sslibdsse.SignerVerifier, stage tuf.HookStage, hookName, filePath string, environment tuf.HookEnvironment, modules, principalIDs []string, signCommit bool, opts ...trustpolicyopts.Option) error {
```

Comment:
I don't love that this is in a random file on disk somewhere...

Reply:
I wonder if the API should take an io.Reader or []byte so we can build better UX later. For now, maybe the cmd package passes an os.File or its contents, but later, we could do neat things like dropping them into a temporary worktree where they can edit the programs with their editor of choice, like any other source file.

Reply:
This should also make testing easier...